src/checks/unordered_key.rs

```diff
@@ -188,4 +188,15 @@ mod tests {
 
         run_unordered_tests(asserts);
     }
+
+    #[test]
+    fn one_unordered_key_with_control_comment_test() {
+        let asserts = vec![
+            (line_entry(1, 3, "FOO=BAR"), None),
+            (line_entry(2, 3, "# dotenv-linter:off LowercaseKey"), None),
```

Comment:
Please also add a test for usual comments.

Reply:
About usual comments.

In the `UnorderedKey` fixer we now handle a significant line (with key) with all previous comments as a whole, when we sort the keys.

E. g.:
```
B=C
# Comment about A variable
A=B
```
will be fixed to:
```
# Comment about A variable
A=B
B=C
```

I will have to remove this behavior, if we use usual comments as group separators.

Is this OK?

Reply:
I think yes.

Reply:
@mgrachev I think this is ambiguous.
This question was raised earlier, as far as I know.

It seems to me not entirely correct to use simple comments as a separator.

For example, if we use comments as separator 
```
DIR=opt
# Comment to FOO line: FOO is BAR
FOO=BAR
BAR=FOO

```
this option will be fixed as:
```
DIR=opt
# Comment to FOO line: FOO is BAR
BAR=FOO
FOO=BAR

```

That is, a comment intended for a specific line will lose its meaning.

Yes, this also can be
ambiguous if comment really intended for group of lines, but user can separate group comment via blank line.

But if we will use usual comments as a separator, we will have trouble with comments to specific lines

Reply:
@DDtKey I think you are right 👍 I missed this case 🤔

Reply:
@DDtKey What do you think about splitting lines into groups using control comments?

Reply:
> @DDtKey What do you think about splitting lines into groups using control comments?

I think this is more of a necessity, we cannot allow reordering of lines beyond the boundaries of comments.

In fact, using the control comments, the user creates the group himself, with additional conditions to processing (ignore specific checks)